ruma-federation-api/src/transactions/send_transaction_message/edu.rs

```diff
@@ -0,0 +1,294 @@
+//! [PUT /_matrix/federation/v1/send/{txnId}](https://matrix.org/docs/spec/server_server/r0.1.3#put-matrix-federation-v1-send-txnid)
+
+use std::collections::BTreeMap;
+
+use js_int::UInt;
+use ruma_common::{encryption::DeviceKeys, presence::PresenceState};
+use ruma_events::{from_raw_json_value, receipt::Receipt};
+use ruma_identifiers::{DeviceIdBox, EventId, RoomId, UserId};
+use serde::{de, Deserialize, Serialize};
+use serde_json::{value::RawValue as RawJsonValue, Value as JsonValue};
+
+// There is one more edu_type synapse recognizes with the note:
+// FIXME: switch to "m.signing_key_update" when MSC1756 is merged into the
+// spec from "org.matrix.signing_key_update"
+
+/// Type for passing ephemeral data to homeservers.
+#[derive(Clone, Debug, Serialize)]
+#[serde(tag = "edu_type", content = "content")]
+pub enum Edu {
```

Comment:
I recommend still making this non-exhaustive, as a "custom" type could suddenly be upgraded to a "recognised" one.

Reply:
Ooo good catch thanks will do!